source/extensions/http/cache/file_system_http_cache/file_system_http_cache.h

```diff
@@ -21,21 +22,33 @@ namespace FileSystemHttpCache {
 using ConfigProto =
     envoy::extensions::http::cache::file_system_http_cache::v3::FileSystemHttpCacheConfig;
 
+class CacheEvictionThread;
+struct CacheShared;
+
+/**
+ * An instance of a cache. There may be multiple caches in a single envoy configuration.
+ * Caches are owned by filter configurations, and jointly own the CacheSingleton; if all
+ * cache instances are destroyed, the CacheSingleton is destroyed.
+ *
+ * See DESIGN.md for details of cache behavior.
+ */
 class FileSystemHttpCache : public HttpCache,
                             public std::enable_shared_from_this<FileSystemHttpCache>,
                             public Logger::Loggable<Logger::Id::cache_filter> {
 public:
-  FileSystemHttpCache(Singleton::InstanceSharedPtr owner, ConfigProto config,
+  FileSystemHttpCache(Singleton::InstanceSharedPtr owner,
+                      CacheEvictionThread& cache_eviction_thread, ConfigProto config,
                       std::shared_ptr<Common::AsyncFiles::AsyncFileManager>&& async_file_manager,
                       Stats::Scope& stats_scope);
+  ~FileSystemHttpCache() override;
 
   // Overrides for HttpCache
   LookupContextPtr makeLookupContext(LookupRequest&& lookup,
                                      Http::StreamDecoderFilterCallbacks& callbacks) override;
   InsertContextPtr makeInsertContext(LookupContextPtr&& lookup_context,
                                      Http::StreamEncoderFilterCallbacks& callbacks) override;
   CacheInfo cacheInfo() const override;
-  const CacheStats& stats() const { return stats_; }
+  const CacheStats& stats() const;
```

Comment:
I think it's OK for the implementation to be inline but it's up to you. Same for config().

Reply:
Can't be inline without significant rejiggering, since the shared class where these values live is declared after the main class. Since the dependencies go both ways, it's a case where moving stuff out of headers is a good option.

Reply:
makes sense; thanks.